cmake/std/atdm/ATDMDevEnvSettings.cmake

```diff
@@ -422,8 +432,15 @@ ENDIF()
 ATDM_SET_ENABLE(TPL_ENABLE_METIS ${TPL_ENABLE_METIS_DEFAULT})
 IF (TPL_ENABLE_METIS)
   ASSERT_DEFINED(ENV{METIS_ROOT})
-  ATDM_SET_CACHE(METIS_INCLUDE_DIRS "$ENV{METIS_ROOT}/include" CACHE FILEPATH)
-  ATDM_SET_CACHE(METIS_LIBRARY_DIRS "$ENV{METIS_ROOT}/lib" CACHE FILEPATH)
+  # allow the user to set metis include and library dirs from the ENV
+  # first set the from the ENV and use (what was here) as the default
+  # then set the cache variable to that.
+  ATDM_SET_ATDM_VAR_FROM_ENV_AND_DEFAULT(METIS_INCLUDE_DIRS
+					 "$ENV{METIS_ROOT}/include")
+  ATDM_SET_CACHE(METIS_INCLUDE_DIRS "${ATDM_METIS_INCLUDE_DIRS}" CACHE FILEPATH)
```

Comment:
Can you revert this and use `export METIS_ROOT=/path/to/metis` instead?

Reply:
There are no new ENVs here.  These variables *should* be supported for consistency. 

This is what causes headaches when a package installs into `lib64` vs `lib` - let us set the correct paths and not depend on `ROOT/lib and ROOT/include` to be the required directories for includes/libs - because it isn't universal.

I can't really see METIS ever needing more than one directory, but who knows.

Reply:
@e10harvey @bartlettroscoe 
I see what you mean in terms of "new" variables.

In my context, I've been skipping all of the `load-env` magic because none of this stuff is really supported.  So, what I have is effectively an ENV that can drive ATDMDevSettings (without needing the framework you all have).

w.r.t. to Evan's comments - I'm looking forward a little bit. Ross has seen the monster I am creating.  It effectively marries `spack` and `ATDMDevSettings`. In that context, I **generate** the env based on what Spack has done. 

It follows then, that I have auto-generated (correct) `INCLUDE_DIRS`, `LIBRARY_DIRS`, and `..._LIBS` (as prescribed by what spack determines the dependencies to be. 

With that in mind, it is natural to define the ATDM_CONFIG api as:
`ATDM_CONFIG_TPL_{INCLUDE_DIRS,LIBRARAY_DIRS,LIBS,LIBRARY_NAMES}`
Which would allow the ENV to be written in a such a way that you could `find_library` if you wanted or hardcode `LIBS` if you want that.

In practice, I am setting `ATDM_CONFIG_foo_INCLUDE_DIRS` and `ATDM_CONFIG_foo_LIBS` - then I can spell out precisely shared or static link lines.

It's a different use case than what SNL is doing right now - but is is kinda a necessary thing if you expect ATDMDevSettings to work on platforms that aren't really well defined (noone can write an ENV file until they sort out the system).

See another comment below (in reply to ross)